app/controllers/users.js

```diff
@@ -9,10 +9,10 @@ exports.registerUser = (req, res, next) => {
   return userService
     .validateUser(user)
     .then(responseUser => {
-      if (responseUser !== null) return Promise.reject(errors.databaseError('user already exists'));
+      if (responseUser) return Promise.reject(errors.invalidData('user already exists'));
```

Comment:
if user exists you must return a different error. For example, an error with status code 409